### PR TITLE
feat: update OpenTelemetry package to 0.2.0-alpha.296

### LINKED ISSUE #236: Is there sample code that runs Telemetry?
I want to verify Telemetry in my environment.
I'm currently testing with CentOS 7.7, not docker.

The readme example code did not work.
Is there any sample code that works?
I would be very happy if I could see how it works with samples/ChatApp.

I think MagicOnion is a great framework!
Thank you.

### PR DESCRIPTION
**Pull request by guitarrapc**
## TL;DR

* This PR offers latest [opentelemetry-dotnet](https://github.com/open-telemetry/opentelemetry-dotnet) API support.
* target version opentelemetry-dotnet 0.2.0-alpha.296

## Summary

As of opentelemetry-dotnet 0.2.0-alpha.296 has broken backward compatibility, MagicOnion.Telemetry also contains many break changes.

**Breaking Changes**

* Removed ITagger, IStats, replace to`MeterFactory` and `TracerFactory`.
* Metric Tags and Tracer tags are refreshed.

**New**

* Add `MagicOnionOpenTelemetryOptions` to support configuration injection.
* Add `IServiceCollection.AddMagicOnionOpenTelemetry` to support Microsoft.Extentions.DependencyInjection.

**Other**

* Update ChatApp.Shared referenced MessagePack to 2.1.15 to support build on docker. 
* Update ChatApp.Unity to use latest MessagePack 2.1.15.
* Add Reconnect Button on ChatApp.Unity to reconnect server when server down during client connection.
* ChatApp.Server.Telemery for first in touch ChatApp and OpenTelemery sample.
* add docker-compose.yaml to support ChatApp.Server run on Docker.
* add docker-compose.telemery.yaml to support ChatApp.Server.Telemery. (**not work until new MagicOnion.Telemery package is released and change project reference to package reference.**)

## Work with

confirm with opentelemetry-dotnet version:

- [x] 0.2.0-alpha.287
- [x] 0.2.0-alpha.296

## API usage sample

New MagicOnion.Telemery offers consistent usage with GenericHost, MagicOnion telemetry will automatically collected for metrics and tracing.

> See sample server implementation **samaples/ChatApp/ChatApp.Server.Telemery**. This offers OpenTelemetry first in touch trial with Prometheus as Metrics and Zipkin as Tracer.

```csharp
await MagicOnionHost.CreateDefaultBuilder()
    .UseMagicOnion()
    .ConfigureServices((hostContext, services) =>
    {
        services.AddMagicOnionOpenTelemetry((options, meterOptions) =>
        {
            // open-telemetry with Prometheus exporter
            meterOptions.MetricExporter = new PrometheusExporter(new PrometheusExporterOptions() { Url = options.MetricsExporterEndpoint });
        },
        (options, tracerBuilder) =>
        {
            // open-telemetry with Zipkin exporter
            tracerBuilder.UseZipkin(o =>
            {
                o.ServiceName = options.ServiceName;
                o.Endpoint = new Uri(options.TracerExporterEndpoint);
            });
        });
        services.AddHostedService<PrometheusExporterMetricsService>();
    })
    .ConfigureServices((hostContext, services) =>
    {
        var meterFactory = services.BuildServiceProvider().GetService<MeterFactory>();
        services.Configure<MagicOnionHostingOptions>(options =>
        {
            options.Service.GlobalFilters.Add(new OpenTelemetryCollectorFilterAttribute());
            options.Service.GlobalStreamingHubFilters.Add(new OpenTelemetryHubCollectorFilterAttribute());
            options.Service.MagicOnionLogger = new OpenTelemetryCollectorLogger(meterFactory);
        });
    })
    .RunConsoleAsync();
```

```json
{
  "MagicOnion": {
    "ServerPorts": [
      {
        "Host": "0.0.0.0",
        "Port": 12345,
        "UseInsecureConnection": true
      }
    ],
    "OpenTelemetry": {
      "ServiceName": "ChatApp.Server.Telemery",
      "MetricsExporterEndpoint": "http://127.0.0.1:9184/metrics/",
      "TracerExporterEndpoint": "http://127.0.0.1:9411/api/v2/spans"
    }
  },
  "Logging": {
    "LogLevel": {
      "Default": "Trace",
      "System": "Information",
      "Microsoft": "Information"
    }
  }
}
```

## REF

closed #236

### CODE REVIEW

File: src/MagicOnion.OpenTelemetry/MagicOnionOpenTelemetryServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+using Microsoft.Extensions.DependencyInjection;
+using OpenTelemetry.Trace.Configuration;
+
+namespace MagicOnion.OpenTelemetry
+{
+    /// <summary>MagicOnion extensions for Microsoft.Extensions.Hosting classes</summary>
+    public static class MagicOnionOpenTelemetryServiceCollectionExtensions
+    {
+        /// <summary>add MagicOnion Telemetry.</summary>
+        public static IServiceCollection AddMagicOnionOptnTelemetry(this IServiceCollection services)
```
Comment (mayuki) [inline src/MagicOnion.OpenTelemetry/MagicOnionOpenTelemetryServiceCollectionExtensions.cs:12]: `AddMagicOnionOptnTelemetry` -> `AddMagicOnionOpenTelemetry`

Comment (guitarrapc) [reply src/MagicOnion.OpenTelemetry/MagicOnionOpenTelemetryServiceCollectionExtensions.cs]: done.
